pkg/shim/shim.go

```diff
@@ -439,7 +444,7 @@ func serve(ctx context.Context, server *ttrpc.Server, signals chan os.Signal, sh
 		return err
 	}
 
-	l, err := serveListener(socketFlag)
+	l, err := serveListener(socketFlag, 3)
```

Comment:
what does 3 mean here?

Reply:
This is the file descriptor. what's happening is that the parent process will create listening socket and pass the FD (3) to the child process. And the child process will starting serving requests on that FD.

Reply:
gotcha! thanks